frontend/src/lib/misc.ts

```
@@ -12,98 +17,127 @@ import type { CityDictItem, CityArrayItem, MaxDistance, IsAmenityOn } from '@/ty
+  [key: string]: number;
+} {
+  const result: { [key: string]: number } = {};
 
-	for (const key in obj) {
-		if (obj.hasOwnProperty(key)) {
-			const snakeCaseKey = key.replace(/([A-Z])/g, '_$1').toLowerCase();
-			result[snakeCaseKey] = obj[key];
-		}
-	}
+  for (const key in obj) {
+    if (obj.hasOwnProperty(key)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)